library/core/class.session.php

```diff
@@ -73,28 +84,92 @@ public function addPermissions($perms) {
         $this->permissions->merge($newPermissions);
     }
 
+    /**
+     * Checks the session user's permission level against the permissions level of the user they want
+     * to perform some action on.
+     *
+     * @param string $sessionPermission
+     * @param int|string $userID
+     * @return bool
+     * @throws NotFoundException Throws exception if there's a problem getting the container.
+     * @throws ContainerException Throws exception if there's a problem getting the container.
+     */
+    public function hasHigherPermissionLevel(string $sessionPermission, $userID) {
```

Comment:
We use the "rank" nomenclature so introducing "level" adds confusion. What about "hasHigherPermissionRank"?